javacutil/src/main/java/org/checkerframework/javacutil/AnnotationUtils.java

```diff
@@ -801,4 +801,26 @@ public static Name getElementValueClassName(
         }
         return annotationSet;
     }
+
+    /**
+     * Returns true if anno is a declaration annotation. In other words, returns true if anno cannot
+     * be written on uses of types.
+     *
+     * @param anno the AnnotationMirror
+     * @return true if anno is a declaration annotation.
+     */
+    public static boolean isDeclarationAnnotation(AnnotationMirror anno) {
```

Comment:
Move this method next to `org.checkerframework.framework.util.AnnotatedTypes.isTypeAnnotation(AnnotationMirror, Class<?>)` or move that one here? It would be better to have them next to each other or have cross-references.

Reply:
`isTypeAnnotation` is never used and it uses a static cache.  How about I just deprecate that method?  I moved `hasTypeQualifierElementTypes` here.

Reply:
Deprecating `isTypeAnnotation` sounds good and this is the more appropriate location for `hasTypeQualifierElementTypes`, so moving here is good.